loopd/swapclient_server.go

```diff
@@ -483,21 +483,50 @@ func (s *swapClientServer) GetLoopInQuote(ctx context.Context,
 		return nil, err
 	}
 
+	var routeHints [][]zpay32.HopHint
+	haveRouteHints := len(req.LoopInRouteHints) != 0
+	haveLastHop := req.LoopInLastHop != nil
 	var lastHop *route.Vertex
-	if req.LoopInLastHop != nil {
+	var lastHopVertex route.Vertex
+
+	if haveLastHop {
 		lastHopVertex, err := route.NewVertexFromBytes(
 			req.LoopInLastHop,
 		)
 		if err != nil {
 			return nil, err
 		}
-
 		lastHop = &lastHopVertex
 	}
 
-	routeHints, err := unmarshallRouteHints(req.LoopInRouteHints)
-	if err != nil {
-		return nil, err
+	switch {
+	// Private and routehints are mutually exclusive as setting private
+	// means we retrieve our own routehints from the connected node.
+	case haveRouteHints && req.Private:
+		return nil, fmt.Errorf("private and route_hints both set")
+
+	case req.Private:
+		// If last_hop is set, we'll only add channels with peers
+		// set to the last_hop parameter
+		includeNodes := make(map[route.Vertex]struct{})
+		if haveLastHop {
+			includeNodes[lastHopVertex] = struct{}{}
+		}
+
+		// Because the Private flag is set, we'll generate our own
+		// set of hop hints
+		routeHints, err = loop.SelectHopHints(
```

Comment:
Got a slightly different implementation here: 
* GetLoopInQuote: selects hop hints and passes them to `impl`
* LoopIn: passes a `Private` bool through to `impl`

Since `swapClientServer` is really just a wrapper for grpc stuff which passes through to our `impl`, I think it would be more consitent to add a `Private` bool to `loop.LoopInQuoteRequest` and pass it through like you do for `LoopIn`?